astropy/config/paths.py

```diff
@@ -27,10 +27,9 @@ def _get_dir_path(rootname: str, cls: type, fallback: str) -> Path:
             path.mkdir(exist_ok=True)
         return path.resolve()
 
-    # first look for XDG_CONFIG_HOME
     if (
-        (xdg_config_home := os.getenv("XDG_CONFIG_HOME")) is not None
-        and (xch := Path(xdg_config_home)).exists()
+        (xdg_dir := os.getenv(f"XDG_{fallback.upper()}_HOME")) is not None
```

Comment:
Might be useful to specify above that `fallback` can take a limited number of values.

Reply:
yes. We could also expose the environment variable directly as a separate argument, since it's a private function, there's no pressure to preserve its API.

Reply:
A good way to do that would be to change the type annotation of the `fallback` parameter from `str` to `Literal["cache", "config"]`, but it would also be a good idea to update the annotation of the `cls` parameter from `type` to `type[_SetTempPath]` and there is more room for improvement elsewhere in this file. My preference would be to only make minimal required changes here to fix the bug and to open a separate pull request for the annotations.

Reply:
Sounds like a plan to me. Final question before I merge: do you want to do the follow up yourself too ? (either way is fine)

Reply:
I wouldn't mind updating the annotations myself.

Reply:
alright then :shipit: